lib/config/configuration.go

```diff
@@ -608,6 +613,12 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 		cfg.AuthServers = append(cfg.AuthServers, cfg.Auth.SSHAddr)
 	}
 
+	// add data_dir to the backend config:
+	if cfg.Auth.StorageConfig.Params == nil {
+		cfg.Auth.StorageConfig.Params = backend.Params{}
+	}
+	cfg.Auth.StorageConfig.Params["data_dir"] = cfg.DataDir
```

Comment:
@klizhentas there are no user-supplied values in "storage config" anymore [1]. Storage config are backend-specific configuration values and they are static-per-backend, with "data_dir" being a reserved word (needs to be reflected in backend developer README). So this behavior is as-designed.

[1] Bolt used to take a random JSON, so that's why you think there are user-supplied values. This is no longer true.